serverless.js

```diff
@@ -18,6 +18,11 @@ class Template extends Component {
 
     return new Proxy(defaultFunction, {
       get: (obj, prop) => {
+        // This handles the weird case when `then` is called on the `defaultFunction`
+        if (prop === 'then') {
```

Comment:
Why is this happening? I would suspect that properties from a promise are bleeding through here. However guarding against `then` does not guarantee that other things may also cause trouble. I suggest that `createCustomMethodHandler` should be renamed `tryCreateCustomMethodHandler` and if a component cannot be found, instead of creating an error, rather just return `obj[prop]`